packages/trace-viewer/src/ui/workbench.tsx

```diff
@@ -67,7 +68,11 @@ export const Workbench: React.FunctionComponent<{
   const activeAction = model ? highlightedAction || selectedAction : undefined;
   const [selectedTime, setSelectedTime] = React.useState<Boundaries | undefined>();
   const [sidebarLocation, setSidebarLocation] = useSetting<'bottom' | 'right'>('propertiesSidebarLocation', 'bottom');
-  const [showRouteActions, , showRouteActionsSetting] = useSetting('show-route-actions', true, 'Show route actions');
+  const [, , showRouteActionsSettingInternal] = useSetting(showRouteActionsSetting ? '' : 'show-route-actions', true, 'Show route actions');
+
+  const showSettings = !showRouteActionsSetting;
+  showRouteActionsSetting ||= showRouteActionsSettingInternal;
+  const showRouteActions = showRouteActionsSetting[0];
```

Comment:
This is all very hard to understand. Would it be easier, if you made the `showRouteActionsSetting` prop required, and always passed it down from every parent component? That'd mean that the `useSetting('show-route-actions', ...)` call is duplicated across multiple components, but it'd make the control flow easier to understand.

Reply:
I think two different props would be wildly easier to understand than this, honestly

Reply:
As discussed offline, filed https://github.com/microsoft/playwright/issues/31863 to track the follow ups.